### ISSUE TITLE
OSX bundle: not building on 10.11 (el capitan)

### ISSUE THREAD

**Issue by ealprr** [quodlibet/quodlibet#1821]
Running ``bundle.sh``:
```bash
Cannot find source to copy: /Users/ealprr/projects/quodlibet-osx/_home/jhbuild_prefix/lib/pango/1.8.0/modules/*.so
```
There is references for pango 1.38.1 in ``quodlibet.modules`` and ``pango-1.0`` during ``build.sh`` but no pango folder in ``_home/jhbuild_prefix/lib/``
```bash
$ find _home/jhbuild_prefix/lib/ -name '*pango*'
_home/jhbuild_prefix/lib//gstreamer-1.0/libgstpango.so
_home/jhbuild_prefix/lib//libpango-1.0.0.dylib
_home/jhbuild_prefix/lib//libpango-1.0.dylib
_home/jhbuild_prefix/lib//libpangocairo-1.0.0.dylib
_home/jhbuild_prefix/lib//libpangocairo-1.0.dylib
_home/jhbuild_prefix/lib//libpangoft2-1.0.0.dylib
_home/jhbuild_prefix/lib//libpangoft2-1.0.dylib
_home/jhbuild_prefix/lib//pkgconfig/pango.pc
_home/jhbuild_prefix/lib//pkgconfig/pangocairo.pc
_home/jhbuild_prefix/lib//pkgconfig/pangoft2.pc
```
I am running El Capitan 10.11.3.
Previous steps (bootstrap and build seems to work).

**Comment by lazka** [comment]
I've only used 10.9 until now. I'm building on 10.11 now..

**Comment by ealprr** [comment]
Well, you may face this issue: https://mail.gnome.org/archives/gtk-osx-devel-list/2014-December/msg00016.html
The workaround works for me. ;)

**Comment by lazka** [comment]
Jup, redoing the build fixed it for me.. at least it's still bulding.

**Comment by lazka** [comment]
hm, multiple issues:

* zlib/bzip2 seem missing and not included in gtk-osx
* we need to build bash first to work around SIP (while it can be disabled in recovery mode OSX just reenabled it for me after a reboot :/)
* gtk-osx bootstrap which is supposed to install bash is broken (and I'm wondering how it ever worked.., but I have a patch)
* the python/libxml2 linking issue. maybe updating python will help there.

I'll look into it but not sure when I find the time.

**Comment by ealprr** [comment]
Which step raise those issues? Because, except for libxml2, they did not appear with my configuration.
Well, at least, before ``bundle.sh``.

**Comment by lazka** [comment]
zlib/bzip2 is just a warning during python configure and results in import errors when running QL.
SIP might be the cause why certain parts of pango aren't build as it breaks configure when run with the system bash.
The bootstrap issue can be worked around.. but I've filed some patches upstream:

https://bugzilla.gnome.org/show_bug.cgi?id=762024
https://bugzilla.gnome.org/show_bug.cgi?id=762025

**Comment by timpulver** [comment]
Any idea why build is failing here?
I’m using Mac OS X 10.11.

```
sh build.sh 
I: unknown keys defined in configuration file: disable_Werror
W: bash has a dependency on unknown "readline" module
*** Checking out xz *** [1/22]
curl --continue-at - -L http://tukaani.org/xz/xz-5.2.1.tar.bz2 -o /Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_checkoutroot/pkgs/xz-5.2.1.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   257  100   257    0     0    200      0  0:00:01  0:00:01 --:--:--   200
  0   360    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
  0   409    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
  0   323    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0
100 1158k  100 1158k    0     0  80211      0  0:00:14  0:00:14 --:--:--  263k
bunzip2 -dc "/Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_checkoutroot/pkgs/xz-5.2.1.tar.bz2" | tar xf -
*** Configuring xz *** [1/22]
./configure --prefix /Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix --libdir '/Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix/lib'   
./configure: line 154: /Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix/bin/bash: No such file or directory
./configure: line 154: exec: /Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix/bin/bash: cannot execute: No such file or directory
*** Error during phase configure of xz: ########## Error running ./configure --prefix /Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix --libdir '/Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix/lib'    *** [1/22]
```

**Comment by lazka** [comment]
I've made some pull request for gtk-osx: https://github.com/jralls/gtk-osx-build/pulls

I'll add the fixes to QL as well for now..

**Comment by lazka** [comment]
pushed now..

remaining issues:

* chromaprint is broken
* there is some cairo warning, not sure what's the problem there as cairo seems to work
* the pango modules

**Comment by timpulver** [comment]
Still some problems for me:

>** Applying patch patches/python2-xcode-stubs.patch *** [3/3]
patch -p1 < "/Users/tim/dev/quodlibet/osx_bundle/modulesets/patches/python2-xcode-stubs.patch"
patching file Lib/distutils/ccompiler.py
Reversed (or previously applied) patch detected!  Assume -R? [n] 



>configure: error: Python module libxml2 is needed to run this package
*** Error during phase configure of itstool: ########## Error running ./configure --prefix /Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix --libdir '/Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix/lib'    *** [8/13]

**Comment by lazka** [comment]
Not 100% sure, but you probably have to start a new build (call ./clean.sh) after the recent changes

**Comment by timpulver** [comment]
I already tried. Hmm…

**Comment by lazka** [comment]
Are you on 10.11 as well?

**Comment by timpulver** [comment]
Yes. 10.11.3

**Comment by lazka** [comment]
ah, copy paste bug

**Comment by lazka** [comment]
fixed now b0ce46a7d8740a0 sorry :/

**Comment by timpulver** [comment]
Getting further, but now I get this:

>*** Checking out gobject-introspection *** [12/69]
curl --continue-at - -L http://ftp.gnome.org/pub/GNOME/sources/gobject-introspection/1.44/gobject-introspection-1.44.0.tar.xz -o /Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_checkoutroot/pkgs/gobject-introspection-1.44.0.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1288k  100 1288k    0     0   296k      0  0:00:04  0:00:04 --:--:--  296k
xzcat -d "/Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_checkoutroot/pkgs/gobject-introspection-1.44.0.tar.xz" | tar xf -
*** Applying patch http://git.gnome.org/browse/gtk-osx/plain/patches/girscanner-objc.patch *** [12/69]
patch -p1 < "/Users/tim/dev/quodlibet/osx_bundle/_home/.cache/jhbuild/girscanner-objc.patch"
patching file giscanner/scannerlexer.l
Hunk #1 succeeded at 126 (offset 18 lines).
Hunk #2 succeeded at 247 with fuzz 2 (offset 47 lines).
*** Error during phase checkout of gobject-introspection: Failed to find patch: gobject-introspection-625195-Fix-non-libtool-linker-flags-on-Darwin.patch *** [12/69]

**Comment by lazka** [comment]
A gtk-osx bug for which I haven't made a pull request yet. selecting "continue" in the jhbuild list should be ok as the patch isn't really needed in our case.

**Comment by timpulver** [comment]
Okay build worked now – `*** success *** [69/69]`!

Now there is another problem. When running `sh bundle.sh` I get this:

```
Cannot find source to copy: /Users/tim/dev/quodlibet/osx_bundle/_home/jhbuild_prefix/lib/pango/1.8.0/modules/*.so
```

**Comment by lazka** [comment]
This is the issue reported in the first post here.

You still can `source env.sh`;  `jhbuild shell`; `../quodlibet/quodlibet.py`

**Comment by timpulver** [comment]
Great, thanks!

**Comment by lazka** [comment]
All fixes got merger upstream now.

Re pango, it turned out that pango removed modules in newer versions and the bundler can't handle that atm. I've posted a proposal to the gtk-osx mailinglist for now: https://mail.gnome.org/archives/gtk-osx-users-list/2016-February/msg00026.html

**Comment by ealprr** [comment]
Many thanks for all your work on this issue!

**Comment by lazka** [comment]
Everything is upstream now.

I've updated "bootstrap.sh" to allow running it again with a working environment, so just call it again instead of starting a new build (but I haven't tested this..)

Only remaining issue: libchromaprint doesn't load

**Comment by lazka** [comment]
Another issue: openssl is no longer included with 10.11, we need to ship our own.

**Comment by lazka** [comment]
openssl/cairo/chromparint are fixed now
